backend/app/services/chat/task_default_knowledge_bases.py

```
@@ -44,13 +51,74 @@ def _build_task_knowledge_base_ref(
+        fallback_user_id=task.user_id,
+    )
+
+
+def _get_default_knowledge_base(db: Session, knowledge_base_id: int) -> Kind | None:
+    """Load an active knowledge base by ID."""
+    return (
+        db.query(Kind)
+        .filter(
+            Kind.id == knowledge_base_id,
+            Kind.kind == "KnowledgeBase",
+            Kind.is_active == True,
```

Ruff (0.15.18)

[error] Avoid equality comparisons to `True`; use `Kind.is_active:` for truth checks

Replace with `Kind.is_active`

(E712)